### PR TITLE
Shape and Layer Messages

### LINKED ISSUE #183: Determine and implement shape messages for visualizer
We have to determine the types of shape we want to draw and the types of ROS messages will we be using to send them to the visualizer. Once this is determined, we should implement our decisions in the Logger like singleton defined in issue #170

### PR DESCRIPTION
**Pull request by FSXAC**
### Description
- Added shape and layer message structure for visualizer communications
- Updated CMake file to include new messages

### Testing Done
Compile passed

### Resolved Issues
Resolves #183 

### Review Checklist
*(Please check every item to indicate your code complies with it (by changing `[ ]`->`[x]`). This will hopefully save both you and the reviewer(s) a lot of time!)*
***It is the reviewers responsibility to also make sure every item here has been covered***
- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/thunderbots_msgs/msg/DrawShape.msg [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# If "rect":   [x, y, width, height]
+# If "poly":   [x1, y1, x2, y2, ...]
+# If "arc":    [center_x, center_y, radius, theta_start, theta_end]
+# If "line":   [x1, y1, x2, y2]
+float64[] data
+
+# Fill of the shape
+string fill
+
+# Stroke of the shape
```
Comment (MathewMacDougall) [inline src/thunderbots_msgs/msg/DrawShape.msg]: Can we list the valid values for the stroke here too, like with `type`?

File: src/thunderbots_msgs/msg/DrawShape.msg [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# Data associated with the shape; expect different form depending on type:
+# If "ellipse": [center_x, center_y, r1, r2]
+# If "rect":   [x, y, width, height]
+# If "poly":   [x1, y1, x2, y2, ...]
+# If "arc":    [center_x, center_y, radius, theta_start, theta_end]
+# If "line":   [x1, y1, x2, y2]
+float64[] data
+
+# Fill of the shape
```
Comment (MathewMacDougall) [inline src/thunderbots_msgs/msg/DrawShape.msg]: I know that this string would be the name of a color from our previous discussions, but could we add a quick comment saying that here too?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Small stuff, otherwise looks good!

File: src/thunderbots_msgs/msg/DrawLayer.msg [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# This message is for basic communication from AI to the visualizr
+# regarding shape drawing
+
+string name
+DrawShape[] shapes
```
Comment (garethellis0) [inline src/thunderbots_msgs/msg/DrawLayer.msg]: nit; github complaining about end-of-line

File: src/thunderbots_msgs/msg/DrawShape.msg [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+# Stroke of the shape
+string stroke
+
+# Weight of the stroke
+uint8 stroke_weight
+
+# Transformations
+float64 transform_rotation
+float64 transform_scale
```
Comment (garethellis0) [inline src/thunderbots_msgs/msg/DrawShape.msg]: nit; github complaining about end-of-file

**Review by garethellis0 — CHANGES_REQUESTED** [review]

File: src/thunderbots_msgs/msg/DrawShape.msg (hunk 1 above)
Comment (FSXAC) [reply src/thunderbots_msgs/msg/DrawShape.msg]: I'm not sure, that entirely depends on how front end interprets the colors.

Comment (MathewMacDougall) [reply src/thunderbots_msgs/msg/DrawShape.msg]: Oh I didn't realize this was also a color value. Your new comments clear this up, so I have no problem with this :+1:

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1: This looks really good (branch will need updating though)

**Comment by MathewMacDougall** [comment]
@garethellis0 I think your comments are resolved?

**Review by qgolsteyn — APPROVED** [review]
LGTM :+1:

**Comment by garethellis0** [comment]
@FSXAC  - just as a note for the future, could you please refrain from resolving comments, and instead leave a quick comment reply explaining what you changed? When I come back to PR's I use unresolved comments as a guide for re-reviewing, which helps give me context again and makes reviewing a lot faster.

(There was absolutely _no_ expectation that you know any of the above, so no worries at all! We should probably try to formalize this as a team at some point)

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Comment by FSXAC** [comment]
> @FSXAC - just as a note for the future, could you please refrain from resolving comments, and instead leave a quick comment reply explaining what you changed? When I come back to PR's I use unresolved comments as a guide for re-reviewing, which helps give me context again and makes reviewing a lot faster.
> 
> (There was absolutely _no_ expectation that you know any of the above, so no worries at all! We should probably try to formalize this as a team at some point)

Yeah absolutely. I didn't know GitHub actually hides them. I thought it's like Atlassian's but I'll do that next time.

**Pull request merged**